docs/reference/indices/upgrade.asciidoc

```diff
@@ -1,9 +1,59 @@
 [[indices-upgrade]]
 == Upgrade
 
-The upgrade API allows to upgrade one or more indices to the latest format
-through an API. The upgrade process converts any segments written
-with previous formats.
+The upgrade API allows to upgrade one or more indices to the latest Lucene
+format through an API. The upgrade process converts any segments written with
+older formats.
+
+.When to use the `upgrade` API
+**************************************************
+
+Newer versions of Lucene often come with a new index format which provides bug
+fixes and performance improvements. In order to take advantage of these
+improvements, the segments in each shard need to be rewritten using the latest
+Lucene format.
+
+.Automatic upgrading
+
+Indices that are actively being written to will automatically write new
+segments in the latest format.  The background merge process which combines
+multiple small segments into a single bigger segment will also write the new
+merged segment in the latest format.
+
+.Optional manual upgrades
+
+However, some old segments may never be merged away because they are already
+too big to be worth merging, and indices that no longer receive changes will
+not be upgraded automatically.  Upgrading segments is not required --
+Elasticsearch can read older formats from the current and previous major
+version of Lucene.
+
+However, you can choose to upgrade old segments manually to take advantage of
```

Comment:
"You can, however,"

It'd be nice to have a link to a page here that describes what is new about the index format in each new Elasticsearch version so people know what advantages they are getting.